rust/ql/src/queries/unusedentities/UnusedVariable.qll

```diff
@@ -16,6 +16,20 @@ predicate isUnused(Variable v) {
   not exists(v.getInitializer())
 }
 
+/**
+ * A callable for which we have incomplete information, for example because an unexpanded
+ * macro call is present. These callables are prone to false positive results from unused
+ * entities queries, unless they are excluded from results.
+ */
+class IncompleteCallable extends Callable {
```

Comment:
Given that we don't use `IncompleteCallable` as a type, this would be simpler as a predicate instead of a class (unless we expect to use it as a type in the future).

Reply:
I went with a `class` because that expresses what we're doing here - saying that an `IncompleteCallable` is a type of `Callable` - not because we need to define member predicates.  This could certainly be done as a predicate (`isIncompleteCallable`) instead, it would be fewer lines of code, but I think readability is (marginally) better how it is.

Reply:
@paldepind you marked this as "requested changes" which is blocking merging.  Please could you either remove this status or convince me I should make this change?

Reply:
Sorry, I've resolved it now. I guess my preference would still be towards using a predicate, but let's keep it as is :)